packages/truffle-debugger/lib/solidity/selectors/index.js

```diff
@@ -321,6 +323,21 @@ let solidity = createSelectorTree({
         node.expression.expression !== undefined &&
         (isContract(node.expression.expression) ||
           isContractType(node.expression.expression))
+    ),
+
+    /**
+     * solidity.current.needsFunctionDepthWorkaround
+     * HACK
+     * Determines if the solidity version used for the contract about to be
+     * called was <0.5.1, to determine whether to use the above workaround
+     * Only call this if the current step is a call or create!
+     */
+    needsFunctionDepthWorkaround: createLeaf(
+      [evm.current.step.callContext],
+      context =>
+        context.compiler !== undefined && //would be undefined for e.g. a precompile
+        context.compiler.name === "solc" &&
+        semver.satisfies(context.compiler.version, "<0.5.1")
```

Comment:
I love using the `semver` library to compare semver values/constraints

Reply:
What, you think I want to make that comparison myself? :P